exporters/committime/collector_bitbucket.py

```diff
@@ -147,22 +141,86 @@ def get_commit_time(self, metric):
         return metric
 
     def get_commit_information(
-        self, api_pattern, api_server, group, project_name, sha, metric
-    ):
-        """Makes an API call to get the commit information"""
-        # Finally, make the API call
+        self,
+        api_pattern: str,
+        api_server: str,
+        group: str,
+        project_name: str,
+        sha: str,
+        metric: CommitMetric,
+    ) -> Optional[dict]:
+        """
+        Call the bitbucket API to get commit information.
+
+        Returns None if any of the following occur:
+        - there was an exception,
+        - the response status code was not a success
+        - the response body was not valid JSON
+        - the response body was JSON, but not a dictionary
+
+        Returns the JSON body otherwise.
+
+        You may assume all of these cases have already been logged.
+        """
         api_response = None
         try:
-            # build the API path using, group, project and commit sha
             path = api_pattern.format(group=group, project=project_name, commit=sha)
-            # create the full URL
             url = pelorus.url_joiner(api_server, path)
-            # send a GET
+
             response = self.__session.request(
                 "GET", url=url, headers=self.DEFAULT_HEADERS
             )
             response.encoding = "utf-8"
-            api_response = response
+            response.raise_for_status()
```

Comment:
The original code checked if the response was exactly 200, while this just checks if it's `400 <= status <= 599`.

I can't think of a case where this matters, but it's worth pointing out.